src/BloomBrowserUI/bookEdit/bloomField/bloomField.ts

```diff
@@ -0,0 +1,254 @@
+    // is on the English qtip (in the FF inspector, the qtip block highlights when you type). https://jira.sil.org/browse/BL-786
+    // This bug mentions the cursor being in the wrong place: https://bugzilla.mozilla.org/show_bug.cgi?id=904846
+    static PrepareNonParagraphField(field:HTMLElement) {
+        if ($(field).text() === '') {
+            //add a span with only a zero-width space in it
+            //enhance: a zero-width placeholder would be a bit better, but libsynphony doesn't know this is a space: //$(this).html('<span class="bloom-ui">&#8203;</span>');
+            $(field).html('&nbsp;');
+            //now we tried deleting it immediately, or after a pause, but that doesn't help. So now we don't delete it until they type or paste something.
+            // REMOVE: why was this doing it for all of the elements? $(container).find(".bloom-editable").one('paste keypress', FixUpOnFirstInput);
+            $(field).one('paste keypress', this.FixUpOnFirstInput);
+        }
+    }
+
+    //In PrepareNonParagraphField(), to work around a FF bug, we made a text box non-empty so that the cursor would should up correctly.
+    //Now, they have entered something, so remove it
+    static FixUpOnFirstInput(event: any) {
+        var field = event.target;
+        //when this was wired up, we used ".one()", but actually we're getting multiple calls for some reason,
+        //and that gets characters in the wrong place because this messes with the insertion point. So now
+        //we check to see if the space is still there before touching it
+        if ($(field).html().indexOf("&nbsp;") === 0) {
+            //earlier we stuck a &nbsp; in to work around a FF bug on empty boxes.
+            //now remove it a soon as they type something
+
+
+            // this caused BL-933 by somehow making us lose the on click event link on the formatButton
+            //   $(this).html($(this).html().replace('&nbsp;', ""));
+
+            //so now we do the follow business, where we select the &nbsp; we want to delete, momements before the character is typed or text pasted
+            var selection = window.getSelection();
+
+            //if we're at the start of the text, we're to the left of the character we want to replace
+            if (selection.anchorOffset === 0) {
+                selection.modify("extend", "forward", "character");
+                //REVIEW: I actually don't know why this is necessary; the pending keypress should do the same thing
+                //But BL-952 showed that without it, we actually somehow end up selecting the format gear icon as well
+                selection.deleteFromDocument();
+            } //if we're at position 1 in the text, then we're just to the right of the character we want to replace
+            else if (selection.anchorOffset === 1) {
+                selection.modify("extend", "backward", "character");
+            }
+        }
+    }
```

Comment:
Do we need code to put the nbsp back in if they delete everything?

Also, the logic here seems to almost duplicate logic above (for paragraph fields?) that just selects the nbsp after inserting it, so the user automatically replaces it himself by typing, pasting, etc. Can we use the same approach (and code) in both places? If not it would be good for something, maybe the overall file comment, to explain why.